src/ucp/wireup/select.c

```diff
@@ -245,42 +288,38 @@ ucp_wireup_select_transport(const ucp_wireup_select_params_t *select_params,
                             int show_error,
                             ucp_wireup_select_info_t *select_info)
 {
-    ucp_ep_h ep           = select_params->ep;
-    ucp_worker_h worker   = ep->worker;
-    ucp_context_h context = worker->context;
-    const ucp_address_entry_t *address_list;
-    unsigned address_count;
+    ucp_ep_h ep                          = select_params->ep;
+    ucp_worker_h worker                  = ep->worker;
+    ucp_context_h context                = worker->context;
+    const ucp_address_entry_t *addr_list = select_params->address_list;
+    unsigned addr_count                  = select_params->address_count;
+    unsigned addr_index;
     uct_tl_resource_desc_t *resource;
     const ucp_address_entry_t *ae;
-    ucp_rsc_index_t rsc_index, best_rsc_index;
-    double score, best_score;
+    ucp_rsc_index_t rsc_index;
     char tls_info[256];
     char *p, *endp;
     uct_iface_attr_t *iface_attr;
     uct_md_attr_t *md_attr;
     uint64_t addr_index_map;
-    unsigned addr_index, best_dst_addr_index;
-    int reachable;
-    int found;
-    uint8_t priority, best_score_priority;
-
-    found               = 0;
-    best_score          = 0.0;
-    best_rsc_index      = 0;
-    best_dst_addr_index = 0;
-    best_score_priority = 0;
-    p                   = tls_info;
-    endp                = tls_info + sizeof(tls_info) - 1;
-    tls_info[0]         = '\0';
-    tl_bitmap          &= select_params->tl_bitmap;
-    show_error          = (select_params->show_error && show_error);
-    address_list        = select_params->address_list;
-    address_count       = select_params->address_count;
+    ucp_wireup_select_info_t sinfo;
```

Comment:
`sinfo = {0};` and move up and remove `memsert()`

Reply:
See comment near `memset`, I tried to document that only initializing `sinfo.reachable = 0` is important, other fields could be uninitialized but compilers warn about this.